src/lib.rs

```diff
@@ -25,61 +24,54 @@ use crate::ip_who::IpWhoIs;
 use crate::us_ofac::validate_country_code;
 
 fn get_providers() -> Vec<Box<dyn LocationProvider>> {
-  let mut providers: Vec<Box<dyn LocationProvider>> = vec![];
+    let mut providers: Vec<Box<dyn LocationProvider>> = vec![];
 
-  #[cfg(feature = "ip_info_provider")]
-  providers.push(Box::new(IpInfoIoFetch {}));
+    #[cfg(feature = "ip_info_provider")]
+    providers.push(Box::new(IpInfoIoFetch {}));
 
-  #[cfg(feature = "ip_who_provider")]
-  providers.push(Box::new(IpWhoIs {}));
+    #[cfg(feature = "ip_who_provider")]
+    providers.push(Box::new(IpWhoIs {}));
 
-  providers
+    providers
 }
 
 // Note: rejected options
 // https://icanhazip.com/ - returns only IP, not enough
 // https://api.iplocation.net/?ip=8.8.8.8 - country only, no region
 // https://ipbase.com/ - requires free plan sign-up
 
-///
-///
-pub struct ComplianceChecker {
-  config: Option<Configuration>,
+/// Main object responsible for validating the country and region
+pub struct ComplianceChecker<'a> {
+    config: Option<&'a Configuration>,
```

Comment:
Sorry I meant to keep compliance checker as a move of the configuration and then use `as_ref()` at the location call
```suggestion
pub struct ComplianceChecker {
    config: Option<Configuration>,
```